JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeRoute.kt

```diff
@@ -108,66 +116,64 @@ fun HomeRoute(
     }
 
     BoxWithConstraints {
-        val useListDetail = when (getWindowSize(maxWidth)) {
-            WindowSize.Compact, WindowSize.Medium -> false
-            WindowSize.Expanded -> true
-        }
-
         // Determine which type of the home screen to display
-        val homeScreenType = if (useListDetail) {
-            HomeScreenType.ListArticle
-        } else {
-            when (uiState) {
-                is HomeUiState.HasPosts -> if (uiState.isArticleOpen) {
-                    HomeScreenType.Article
-                } else {
-                    HomeScreenType.List
+        val homeScreenType = when (getWindowSize(maxWidth)) {
```

Comment:
We should only calculate the `windowSize` whenever the `maxWidth` changes. What about including this in a `remember`?

Reply:
Updated! This seems like it shouldn't make too much of a difference, since `getWindowSize` should be pretty cheap.